### PR TITLE
Fix ntasks job option,work_dir job option and slurmdb_jobs db connection

### PR DESCRIPTION
**Pull request by ahmedhany93**
We have edited pyslurm.pyx file to fix the ntasks job option which was commented out, fix the work_dir job option which was not implemented and to fix slurmdb_jobs_get() database connection issue. There was no statement to close the database connection after fetching data, and as a result, the function was unable to fetch changes to the database running on flask server.

### CODE REVIEW

**Comment by ahmedhany93** [comment]
It seems that the master branch already failed in the same steps.
Any way, I just wanted to ask if there was a reason for commenting out the edited options?
Also was there a reason for not closing the db_conn in slurdb_jobs_get ?

File: pyslurm/pyslurm.pyx [hunk 1]
```diff
@@ -2833,8 +2837,13 @@ cdef class job:
 
         # FIXME: should this be python's getcwd or C's getcwd?
         # also, allow option to specify work_dir, if not, set default
-        cwd = os.getcwd().encode("UTF-8", "replace")
-        desc.work_dir = cwd
+
+        if job_opts.get("work_dir"):
```
Comment (giovtorres) [inline pyslurm/pyslurm.pyx]: :+1:

File: pyslurm/pyslurm.pyx [hunk 2]
```diff
@@ -5592,6 +5601,7 @@ cdef class slurmdb_jobs:
             dict J_dict = {}
             slurm.List JOBSList
             slurm.ListIterator iters = NULL
+            void* db_conn = slurm.slurmdb_connection_get()
```
Comment (giovtorres) [inline pyslurm/pyslurm.pyx]: :+1:  I can't remember if this got taken out or was missing to begin with.

File: pyslurm/pyslurm.pyx [hunk 3]
```diff
@@ -2746,6 +2746,10 @@ cdef class job:
 #            desc.min_cpus = job_opts.get("ntasks")
 
         # TODO: ntasks_set, cpus_set
+
+        if job_opts.get("ntasks"):
+            desc.min_cpus = job_opts.get("ntasks") * job_opts.get("cpus_per_task")
```
Comment (giovtorres) [inline pyslurm/pyslurm.pyx]: Is this right?  I think there's some conditions to calculating min_cpus.

**Comment by giovtorres** [comment]
The Python 2.6 tests need to get fixed. :disappointed:

**Comment by giovtorres** [comment]
Can you rebase please?

**Comment by giovtorres** [comment]
No response. Please rebase. Closing for now.

**Pull request closed without merging**